Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1

```diff
@@ -0,0 +1,356 @@
+            }
+            else {
+                $cloudServiceWithDisabledRDPAccess += $_ | Select-Object @{Expression={($resource.ResourceGroupName)};Label="ResourceGroupName"},@{Expression={$resource.Name};Label="CloudServiceName"}
+            }
+        }   
+    }
+    catch {
+        Write-Host "Error occured while checking config of cloud service(s). ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+        break
+    }
+    
+    $totalCloudServiceWithEnableRDPAccess = ($cloudServiceWithEnabledRDPAccess | Measure-Object).Count
+    $totalCloudServiceWithDisabledRDPAccess = ($cloudServiceWithDisabledRDPAccess | Measure-Object).Count
+    
+    Write-Host "Cloud service(s) with enabled RDP access: [$($totalCloudServiceWithEnableRDPAccess)]"
+    Write-Host "Cloud service(s) with disabled RDP access: [$($totalCloudServiceWithDisabledRDPAccess)]"
+    Write-Host "`t"
+
+    # Performing remediation on cloud service(s).
+    try
+    {
+        if ($totalCloudServiceWithEnableRDPAccess -gt 0)
+        {
+            # Creating the log file
+            Write-Host "Backing up config of cloud service(s) detail." -ForegroundColor $([Constants]::MessageType.Info)
+            $cloudServiceWithEnabledRDPAccess | ConvertTo-json | out-file "$($folderpath)\CloudServiceWithEnabledRDPAccess.json"  
+            Write-Host "Path: $($folderpath)\CloudServiceWithEnabledRDPAccess.json"     
+            
+            # Asking user to verify logs and select 'Y' to proceed
+            if(-not $Force)
+            {
+                Write-Host "Do you want to disable RDP access from cloud service(s) listed in above path?" -ForegroundColor Yellow -NoNewline
+                $UserInput = Read-Host -Prompt "(Y|N)"
+                if($UserInput -ne "Y")
+                {
+                    return;
+                }
+            }
+            
+            Write-Host "`t"
+            Write-Host "Disabling RDP access on [$($totalCloudServiceWithEnableRDPAccess)] cloud service(s)..."
+            try 
+            {
+                $cloudServiceWithEnabledRDPAccess | ForEach-Object {
+                    $serviceName = $_.CloudServiceName
+                    $rgName = $_.ResourceGroupName
+                    
+                    # Disabling RDP access
+                    $_.RDPExtensionDetails | ForEach-Object {
+                        $output = Remove-AzureServiceRemoteDesktopExtension -ServiceName $serviceName -Slot $_.Slot -Role $_.Role -ErrorAction SilentlyContinue
+                        if($null -eq $output -and $output.OperationStatus -ine "Succeeded")
+                        {
+                            $item =  New-Object psobject -Property @{  
+                                CloudServiceName = $serviceName                
+                                ResourceGroupName = $rgName
+                            }
+    
+                            $skippedCloudServiceFromRemediation += $item
```

Comment:
Are you breaking on first failure?

Reply:
We have used 'For' loop to remove extensions from cloud service(s), so if we failed to remove any extension(due to permission issue or cmd. failure) then we are breaking flow of removing other extension of that cloud service and simply storing service and RG name in an array(containing list of cloud services skipped from remediation).